pandas/core/internals/blocks.py

```diff
@@ -1582,6 +1582,50 @@ def _replace_coerce(
                 )
         return self
 
+    def _discriminate_replace_and_coerce_block(
+        self,
+        mask: np.ndarray,
+        new_rb: "Block",
+        to_replace: Any,
+        value: Any,
+        inplace: bool,
+        convert: bool,
+        regex: bool,
+    ) -> "Block":
```

Comment:
looks like this might actually be `List["Block"]`?

Reply:
Yes, good catch, thanks (same with `new_rb`)